src/content/hinting.ts

```diff
@@ -716,7 +716,11 @@ export function hintByText(match) {
             } else {
                 text = hint.textContent
             }
-            return text.indexOf(match) >= 0
+            if (match.constructor.name == "RegExp") {
+                return text.match(match) !== null
+            } else
+                return text.indexOf(match) >= 0
+            }
```

Comment:
use `text.search(match) >= 0` instead. It works for both regexp and string patterns.

Reply:
I have tried that and it seems that it always interprets the text as a regex. That is
`:hint -f $`
matches all links, not only those containing '$'. This seems to confirm it: https://www.w3schools.com/jsref/jsref_match.asp

Reply:
Sorry, my bad there.

Please use `match instanceof RegExp` for the check and `text.includes(match)` for the string match.

Happy to merge then.

Reply:
Thanks for the tips, it's the first time I'm touching typescript. I have also added some comment above the function.